lib/facility.rb

```diff
@@ -1,14 +1,81 @@
+require "date"
+
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, 
+                :address, 
+                :phone, 
+                :services,
+                :hours, 
+                :registered_vehicles, 
+                :collected_fees,
+                :holidays_closed
+
+
+  LIST_OF_SERVICES = ["New Drivers License", 
+                      "Renew Drivers License", 
+                      "Vehicle Registration", 
+                      "Written Test",
+                      "Road Test"]
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(input)
+    @name = input[:name]
+    @address = input[:address]
+    @phone = input[:phone]
+    @hours = input[:hours]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    if input[:holidays_closed] != nil
+      @holidays_closed = input[:holidays_closed]
+    end
   end
 
-  def add_services(service)
-    @services << service
+  def add_service(service)
+    if LIST_OF_SERVICES.include?(service)
+      @services << service
+      @services
+    end
   end
+
+  def register_vehicle(vehicle)
+    vehicle.registration_date = Date.today
+    if vehicle.registered == false
+      if vehicle.engine == :ev
+        @collected_fees += 200
+        vehicle.plate_type =:ev
+      elsif vehicle.antique? == true
+        @collected_fees += 25
+        vehicle.plate_type = :antique
+      else
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+      end
+    @registered_vehicles << vehicle
+    vehicle.registered = true
+    vehicle
+    end
+  end
```

Comment:
We will want to test that the facility has the "Vehicle Registration" service before we do anything to that vehicle object. The rest of the logic in this method looks good though!